homeassistant/components/elgato/light.py

```diff
@@ -36,6 +37,16 @@ async def async_setup_entry(
     info = await elgato.info()
     async_add_entities([ElgatoLight(elgato, info)], True)
 
+    platform = current_platform.get()
+    if not platform:
```

Comment:
Is there a situation where `platform` can be `None`? I have had similar code in other (custom) integrations, but I never checked if platform is set.

Reply:
This is needed for typing, as get() could return None according to the result type.

Reply:
It's a context var and so it can be None. It should never be None in HA.

Reply:
We could introduce a proxy method around this? One that returns the expected type, and raises when it is None?

Reply:
Added #50014

Reply:
Adjusted.